src/main/kotlin/minesweeper/entity/MineField.kt

```diff
@@ -20,7 +20,51 @@ class MineField(
 
     fun countAroundMines(coordinate: Coordinate): Int {
         return coordinate.adjacentCoordinates()
-            .filter { it.isWithinBounds(width.value, height.value) }
-            .count { _cells.findCell(it) is Cell.Mine }
+            .filter { it.isWithinBounds(width, height) }
+            .count { !_cells.findCell(it).isSafe() }
+    }
+
+    fun open(coordinate: Coordinate) {
+        val cell = _cells.findCell(coordinate)
+        if (cell.isRevealed) return
+
+        cell.open()
+        if (shouldOpenAdjacentCells(cell)) {
+            openAdjacentEmptyCells(coordinate)
+        }
+    }
+
+    private fun shouldOpenAdjacentCells(cell: Cell): Boolean {
+        return cell.isSafe() && countAroundMines(cell.coordinate) == NO_ADJACENT_MINES
+    }
+
+    private fun openAdjacentEmptyCells(coordinate: Coordinate) {
+        coordinate.adjacentCoordinates()
+            .filter { it.isWithinBounds(width, height) && !_cells.findCell(it).isRevealed }
```

Comment:
이 filter부분이 개인적으로 조금 아쉽다고 생각이 드는데요. 🤔
`cells` 객체에게  특정 `coordinate`의 `adjacentCoordinates`를 찾아줘! 라고 메시지를 보내면 
조금 더 깔끔해지지 않을까요? 

```suggestion
        _cells. adjacentCoordinates(coordinate)
            .filter { !it.isRevealed }
```

지금 생각해보니 adjacentCoordinates 메서드의[네이밍](https://tecoble.techcourse.co.kr/post/2020-04-26-Method-Naming/)이 행위를 드러내주지 못하는 것 같은데, 동사구 네이밍으로 변경해보면 어떨까요? 🤗